File: 1_two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        d=dict()
        for i,elem in enumerate( nums):
            d[elem]=i
        for i,elem in enumerate( nums):
            try:
                _candidate=d[target-elem]
                if _candidate>=0 and _candidate!=i:
                    return [i,_candidate]
            except:
                pass#couldn't find an index that satifies constraint, key error

if __name__ == "__main__":
    
       
    s=Solution()
    
    

File: solutions/hashmap/1_two_sum.cpp
#include <iostream>
#include <vector>
using namespace std;


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        for (int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {
                if (nums[i] + nums[j] == target) {
                    return {i, j};
                }
            }
        }
        return {}; // No solution found
    }
};
int main (){
    Solution *s=new Solution();
    vector<int> v={1,4};
    s->twoSum(v,5);
    for (int i=0;i<v.size();i++){
        cout<<v[i];
    }
    return 0;
}